test/AST-Quake/vector_front_back.cpp

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: cudaq-quake %s | FileCheck %s
+
+// Simple tests of vector front/back on different std::vector<> types.
+
+#include <cudaq.h>
+
+__qpu__ void testFrontFloat() {
+    std::vector<float> vec_float{0.0, 1.0, 2.0, 3.0};
+    auto zero = vec_float.front();
+}
+
+
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__function_testFrontFloat._Z14testFrontFloatv() attributes {"cudaq-entrypoint", "cudaq-kernel", no_this} {
+// CHECK:           %[[VAL_0:.*]] = arith.constant 3.000000e+00 : f32
+// CHECK:           %[[VAL_1:.*]] = arith.constant 2.000000e+00 : f32
+// CHECK:           %[[VAL_2:.*]] = arith.constant 1.000000e+00 : f32
+// CHECK:           %[[VAL_3:.*]] = arith.constant 0.000000e+00 : f32
+// CHECK:           %[[VAL_4:.*]] = cc.alloca !cc.array<f32 x 4>
```

Comment:
nit: These could be `CHECK-DAG:` since the order of the constants is irrelevant.